llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -669,6 +673,65 @@ Value *VPInstruction::generate(VPTransformState &State) {
     }
     return NewPhi;
   }
+  case VPInstruction::CSAMaskPhi: {
+    IRBuilder<>::InsertPointGuard Guard(State.Builder);
+    State.Builder.SetInsertPoint(State.CFG.PrevBB->getFirstNonPHI());
```

Comment:
Why does the builder need to be reset here?

Reply:
addressed